torch/csrc/jit/codegen/cuda/type.cpp

```diff
@@ -160,6 +162,17 @@ DataType getTypeFromComplexType(DataType dtype) {
   }
 }
 
+bool isSupportedType(DataType dtype) {
+  auto prop = at::cuda::getCurrentDeviceProperties();
+  auto major_ver = prop->major;
+  switch (dtype) {
+    case DataType::BFloat16:
+      return major_ver >= 8;
+    default:
```

Comment:
Haven't checked other types. Probably we should also check `half`.

Reply:
`half` seems to be supported since v 5.3, so should be fine.

Reply:
I think half should be supported starting from something like SM35, which is pretty old and no longer supported by PyTorch.